testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/security/SaslLegacyMechanismConfigurationTestCase.java

```diff
@@ -0,0 +1,371 @@
+         ignoreTest = securityRealm == null;
+
+         //if security-realm on htt-remoting-connector is emoty,it means that legacy security is not allowed and this test should be ignored
+         org.junit.Assume.assumeTrue(securityRealm != null);
+
+
+         authentication = execute(Operations.createReadResourceOperation(LOCAL_AUTHENTICATION.append().toModelNode()), mcc);
+         if(SUCCESS.equals(authentication.get(OUTCOME).asString())) {
+            localAuthentication = authentication.get(RESULT);
+         }
+
+         authentication = execute(Operations.createReadResourceOperation(AUTHENTICATION_PROPS.append().toModelNode()), mcc);
+         if(SUCCESS.equals(authentication.get(OUTCOME).asString())) {
+            propsAuthentication = authentication.get(RESULT);
+         }
+
+         authentication = execute(Operations.createReadAttributeOperation(HTTP_REMOTING_CONNECTOR.append().toModelNode(), "sasl-authentication-factory"), mcc);
+         if(SUCCESS.equals(authentication.get(OUTCOME).asString())) {
+            saslAuthenticationFactory = authentication.get(RESULT).asString();
+         }
+         if("undefined".equals(saslAuthenticationFactory)) {
+            saslAuthenticationFactory = null;
+         }
+
+         ModelNode compositeOperation = Operations.createCompositeOperation();
+         if(localAuthentication != null) {
+            compositeOperation.get(ModelDescriptionConstants.STEPS).add(Operations.createRemoveOperation(LOCAL_AUTHENTICATION.toModelNode()));
+         }
+         compositeOperation.get(ModelDescriptionConstants.STEPS).add(addSaslMechanisms());
+         compositeOperation.get(ModelDescriptionConstants.STEPS).add(addPolicyNoanonymous());
+
+         String users = new File(SaslLegacyMechanismConfigurationTestCase.class.getResource("users.properties").getFile()).getAbsolutePath();
+         compositeOperation.get(ModelDescriptionConstants.STEPS).add(Operations.createWriteAttributeOperation(AUTHENTICATION_PROPS.toModelNode(),"path", users));
+         compositeOperation.get(ModelDescriptionConstants.STEPS).add(Operations.createWriteAttributeOperation(AUTHENTICATION_PROPS.toModelNode(),"plain-text", "true"));
+         compositeOperation.get(ModelDescriptionConstants.STEPS).add(Operations.createUndefineAttributeOperation(AUTHENTICATION_PROPS.toModelNode(),"relative-to"));
+
+         if(saslAuthenticationFactory != null) {
+            compositeOperation.get(ModelDescriptionConstants.STEPS).add(Operations.createUndefineAttributeOperation(HTTP_REMOTING_CONNECTOR.append().toModelNode(), "sasl-authentication-factory"));
+         }
+
+         ModelNode response = execute(compositeOperation, mcc);
+         Assert.assertEquals(response.toString(), SUCCESS, response.get(OUTCOME).asString());
+
+         ServerReload.reloadIfRequired(managementClient.getControllerClient());
+      }
+
+      @Override
+      public void tearDown(ManagementClient managementClient, String containerId) throws Exception {
```

Comment:
@JiriOndrusek I think you can get rid of this method if you extend SnapshotRestoreSetupTask instead